src/classes/SuperchainFormFrame.ts

```diff
@@ -0,0 +1,123 @@
+import { FrameMetadataType, getFrameHtmlResponse } from "@coinbase/onchainkit";
+import * as Sentry from "@sentry/nextjs";
+import { z } from "zod";
+
+type HubspotFields = [
+  {
+    name: "email";
+    value: string;
+  },
+  {
+    name: "superchain_chain";
+    value: string;
+  },
+  {
+    name: "website";
+    value: string;
+  },
+  {
+    name: "farcaster_handle";
+    value: string;
+  },
+];
+
+const validChains = z.union([
+  z.literal("base"),
+  z.literal("zora"),
+  z.literal("fraxtal"),
+  z.literal("op-mainnet"),
+  z.literal("mode"),
+]);
+
+const validQueryType = z.union([
+  z.literal("apply"),
+  z.literal("chain"),
+  z.literal("email"),
+  z.literal("website"),
+  z.literal("redirect"),
+]);
+
+export class SuperChainFormFrame {
+  static getQueryType = (type: string) => {
+    return validQueryType.parse(type);
+  };
+
+  static getChain = (chain: string) => {
+    let lowercaseChain = chain.toLowerCase();
+    if (lowercaseChain === "op mainnet") {
+      lowercaseChain = "op-mainnet";
+    }
+    return validChains.parse(lowercaseChain);
+  };
+
+  static getConvertedChain = (chain: string) => {
+    let convertedChain: string;
+
+    switch (chain) {
+      case "base":
+        convertedChain = "Base";
+        break;
+
+      case "fraxtal":
+        convertedChain = "Frax";
+        break;
+
+      case "mode":
+        convertedChain = "Mode";
+        break;
+
+      case "op-mainnet":
+        convertedChain = "Optimism";
+        break;
+
+      case "zora":
+        convertedChain = "Zora";
+        break;
+
+      default:
+        throw new Error("Valid chain not found");
+    }
+    return convertedChain;
+  };
+
+  static getEmail = (email: string) => {
+    return z.string().email().parse(email);
+  };
+
+  static getWebsiteUrl = (url: string) => {
+    const regex =
+      // eslint-disable-next-line no-useless-escape
+      /^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,})([\/\w \.-]*)*\/?$/;
```

Comment:
to validate a url you can just do `new URL(theUrl)` in a try catch

Reply:
Right. I think we also want to accept `thirdweb.com` without `https://`

Reply:
can you just append https:// if not present?

there's a security warning because of that regex

Reply:
Ah yeah good point